certbot/main.py

```diff
@@ -866,6 +867,56 @@ def _post_logging_setup(config, plugins, cli_args):
     logger.debug("Discovered plugins: %r", plugins)
 
 
+def acquire_file_lock(lock_path):
```

Comment:
I can if you prefer. I put it here because it's unlikely people will ever need it but I can move it if you like. If I do, I'll want to expand on it's limitations more with the biggest being you can't use it as a lock between threads.